src/Microsoft.Extensions.PlatformAbstractions.Internal/Native/NativeMethods.Unix.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Extensions.PlatformAbstractions.Internal.Native
+{
+    internal static partial class NativeMethods
+    {
+        public static class Unix
+        {
+            public unsafe static string GetUname()
+            {
+                // Utsname shouldn't be larger than 2K
+                var buf = stackalloc byte[2048];
+
+                try
+                {
+                    if (uname((IntPtr)buf) == 0)
+                    {
+                        return Marshal.PtrToStringAnsi((IntPtr)buf);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    throw new PlatformNotSupportedException("Error reading Unix name", ex);
+                }
+                throw new PlatformNotSupportedException("Unknown error reading Unix name");
+            }
+
+            [DllImport("libc")]
```

Comment:
We spent a lot of time in CoreFX not directly DllImport'ing native functions whose signatures could change from platform to platform (i.e. C++ 'int' could return 64-bit int on some platforms).  IMO - we shouldn't be DllImporting native functions here for the same reasons.

@sokket @nguerrera @stephentoub